src/vm/vm_core.rs

```diff
@@ -996,6 +996,138 @@ impl VirtualMachine {
     }
 }
 
+pub struct VirtualMachineBuilder {
+    pub(crate) run_context: RunContext,
+    pub(crate) builtin_runners: Vec<(String, BuiltinRunner)>,
+    pub(crate) segments: MemorySegmentManager,
+    pub(crate) _program_base: Option<MaybeRelocatable>,
+    pub(crate) memory: Memory,
+    pub(crate) accessed_addresses: Option<Vec<Relocatable>>,
+    pub(crate) trace: Option<Vec<TraceEntry>>,
+    pub(crate) current_step: usize,
+    skip_instruction_execution: bool,
+    run_finished: bool,
+}
+
+impl VirtualMachineBuilder {
+    pub fn new(trace_enabled: bool) -> VirtualMachineBuilder {
```

Comment:
Why is trace enable treated differently than the other fields?
Couldn't it also have it setter method?

Reply:
It also has one but I just reproduced VirtualMachine constructor behavior. Don't know if it's the right way of doing it or if it should be created empty.